app/gather/api/views.py

```diff
@@ -44,3 +59,35 @@ class MaskViewSet(MtViewSetMixin, ModelViewSet):
     search_fields = ('survey__name', 'name', 'columns',)
     ordering = ('survey', 'name',)
     mt_field = 'survey'
+
+
+@api_view(['POST', 'DELETE'])
+@renderer_classes([JSONRenderer])
+@permission_classes([permissions.IsAuthenticated])
+def consumer_config(request, *args, **kwargs):
+    _survey_name = request.data.get('name')
+    _headers = {}
+    if not _survey_name:
+        return Response('Missing survey name', status=status.HTTP_400_BAD_REQUEST)
+
+    _realm = get_current_realm(request)
+    _headers[settings.TENANCY_HEADER] = _realm
+
+    # Configure Consumers
+    if not settings.AUTO_CONFIG_CONSUMERS:
+        return Response('Consumer auto configuration is turned off', status=status.HTTP_200_OK)
+
+    consumer_settings = copy.deepcopy(settings.CONSUMER_SETTINGS)
+    if request.method == 'POST':
+        consumers, errors = configure_consumers(consumer_settings, _survey_name, _headers)
+        if errors:
+            return Response(errors, status=status.HTTP_400_BAD_REQUEST)
+        return Response(f'Configured {consumers} consumers successfully', status=status.HTTP_200_OK)
+    elif request.method == 'DELETE':
+        consumer_settings = copy.deepcopy(settings.CONSUMER_SETTINGS)
+        errors = delete_survey_subscription(consumer_settings, _survey_name, _headers)
+        if errors:
+            return Response(errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
you don't need to duplicate this here if you are not altering the setting (you needed due to the "pop" method but you already removed it)

Reply:
Duplicate the error check? I'm not sure i follow.

Reply:
:joy:  sorry it appeared in the wrong line :man_facepalming: 

`consumer_settings = copy.deepcopy(settings.CONSUMER_SETTINGS)` this is declared twice :grimacing: